### PR TITLE
This patch fixes issue #3258

### PR DESCRIPTION
**Pull request by petrus-v**
I rebase my PR on oca/ocb!

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Seems OK

:+1:

**Comment by ccomb** [comment]
:+1:

**Comment by StefanRijnhart** [comment]
This changes the method signature and breaks anyone's custom or community code that hooks into this API point. We have learned to be very careful with this because OCB is now known to cause instability in this respect unfortunately. :-1:

**Comment by pedrobaeza** [comment]
Do you think this kind of changes in JS are going to affect any custom module?

**Comment by StefanRijnhart** [comment]
Typically a JS module hooks into such methods, right? And then changing the method signature would be causing crashes. Or am I wrong in this respect?

**Comment by pedrobaeza** [comment]
Well, I'm not sure because my knolewdge of JS is not good enough.

Any other can give their advise here?

**Comment by hbrunn** [comment]
JS doesn't really care for method signatures. calling `setup_search_view(only_first_parameter)` results in `options == undefined` as far as the function's code is concerned, and given that https://github.com/OCA/OCB/blob/8.0/addons/web/static/src/js/search.js#L376 assigns sane defaults, I think that's just fine.
tldr: JS doesn't have polymorphic functions

**Comment by StefanRijnhart** [comment]
Great insight, thanks @hbrunn. So in this case it should be alright. I see there is a pending upstream pr: #3266, so :+1: for me and thanks @petrus-v for the patch!

**Comment by petrus-v** [comment]
thanks for merging.

I'm quite new with OCA process, what sould I do to back port this PR on 7.0 branch where the issue is specially true ?

Should I make an other PR against v7.0 or like odoo process an OCA team back port PRs ?

**Comment by pedrobaeza** [comment]
You have to cherry-pick the diff in the 7.0 OCB branch, and make another PR.

**Comment by petrus-v** [comment]
Ok, thanks for your quick reply! I'll do that.
